webui/src/pages/repositories/wizard/wizard.jsx

```diff
@@ -0,0 +1,106 @@
+                    progressBar = false,
+                    isGoBackButton = false,
+                    isSkipButton = false,
+                    onComplete=(any) => any,
+                    canProceed = true,
+                    onNextStep = () => {},
+                    children
+                }) => {
+
+    const [state, setState] = useState({
+        stepWizard: {},
+    });
+
+    const onStepChange = () => {
+        setState({...state});
+        onNextStep();
+    };
+
+    const setInstance = stepWizard => {
+        setState({...state, stepWizard: stepWizard});
+    }
+
+    return (
+        <Container className='container'>
+            <div className='jumbotron'>
+                <Row className={'justify-content-center'}>
+                    <Col>
+                        <StepWizard
+                            onStepChange={onStepChange}
+                            transitions={transitions}
+                            nav={progressBar && <WizardNav />}
+                            instance={setInstance}
+                            isHashEnabled={hashEnabled}
+                        >
+                            {children}
+                        </StepWizard>
+                    </Col>
+                </Row>
+            </div>
+
+            {<WizardController
+                stepWizard={state.stepWizard}
+                canProceed={canProceed}
+                isSkipButton={isSkipButton}
+                isGoBackButton={isGoBackButton}
+                onComplete={onComplete}/>}
+
+        </Container>
+    );
+};
+
+const WizardNav = ({totalSteps, currentStep}) => {
+    return (
+        <ProgressBar className={'wizard-progress-bar'} striped max={totalSteps} min={1} now={currentStep} />
+    );
+}
+
+const WizardController = ({stepWizard, canProceed, isSkipButton = false, onComplete}) => {
+    const finalStep = () => {
+        onComplete();
+    }
+    return (
+        <Container>
+            <Row className={'justify-content-center'}>
+                {
+                    isSkipButton && stepWizard.currentStep < stepWizard.totalSteps ?
+                        <Col className={"col-1 mb-2 mt-2"}>
+                            <button className='btn btn-secondary btn-block' onClick={stepWizard.nextStep}>Skip</button>
+                        </Col>
+                        :
+                        null
+                }
+                <Col className={"col-2 mb-2 mt-2"}>
+                    {
+                        stepWizard.currentStep < stepWizard.totalSteps ?
+                            <div>
+                                <button className='btn btn-primary btn-block' disabled={!canProceed} onClick={stepWizard.nextStep}>Next Step</button>
+                            </div>
+                            :
+                            <button className='btn btn-success btn-block' onClick={finalStep} disabled={!canProceed}>Finish</button>
+                    }
+                </Col>
```

Comment:
Confused by this layout.  There are 2 or 3 buttons.  The first, if present, goes in one column.  The other 2 both go into a single column, separated by a colon.  What is the intended effect of this?  I think it will wrap strangely on a narrow display, and always display strangely (e.g., unequal spacing).

Reply:
It's for the skip button to take less space than the other ones.
I think that the `<div>` there is a left over from before. I'll get rid of it.

The two buttons you're referring to are never displayed together: if `stepWizard.currentStep < stepWizard.totalSteps` is true, the first button will be displayed in this column, if not, the other one will.

The difference between the column defined in line 85 and the column that's defined in line 91 is that the former might not appear at all and latter will always appear.